appengine/src/test/java/com/google/googleinterns/gscribe/services/ExamServiceTest.java

```diff
@@ -0,0 +1,431 @@
+    private final static String spreadsheetId = "spreadsheet_id";
+    private final static String sheetName = "sheet_name";
+    private final static ExamRequest examRequest = new ExamRequest();
+    private final static ValueRange valueRangeForNumberOfQuestions = new ValueRange();
+    private final static ValueRange valueRangeWhole = new ValueRange();
+    private final static List<List<Object>> globalValueRangeDataWhole = new ArrayList<>();
+
+    @Mock
+    SpreadsheetService spreadsheetService;
+    @InjectMocks
+    ExamServiceImpl examService;
+
+    @BeforeAll
+    private static void init() {
+        examRequest.setSheetName(sheetName);
+        examRequest.setSpreadsheetID(spreadsheetId);
+        List<List<Object>> valueRangeDataFoNumberOfQuestions = new ArrayList<>();
+        valueRangeDataFoNumberOfQuestions.add(Arrays.asList(new String[]{"Duration of exam:"}));
+        valueRangeDataFoNumberOfQuestions.add(Arrays.asList(new String[]{"Question Type"}));
+        valueRangeDataFoNumberOfQuestions.add(Arrays.asList(new String[]{"SUBJECTIVE"}));
+        valueRangeDataFoNumberOfQuestions.add(Arrays.asList(new String[]{"MCQ"}));
+        valueRangeForNumberOfQuestions.setValues(valueRangeDataFoNumberOfQuestions);
+        globalValueRangeDataWhole.add(Arrays.asList(new String[]{"Duration of exam:", "100"}));
+        globalValueRangeDataWhole.add(Arrays.asList(new String[]{"Question Type", "Question statement", "Option A", "Option B", "Option C", "Option D", "Points"}));
+        globalValueRangeDataWhole.add(Arrays.asList(new String[]{"SUBJECTIVE", "SubjectiveStatement1", "", "", "", "", "2"}));
+        globalValueRangeDataWhole.add(Arrays.asList(new String[]{"MCQ", "MCQStatement1", "OptionA", "OptionB", "OptionC", "OptionD", "3"}));
+    }
+
+    private List<List<Object>> globalValueRangeDataWholeCopy() {
+        List<List<Object>> copyList = new ArrayList<>();
+        for (List<Object> list : globalValueRangeDataWhole) {
+            List<Object> copyRow = new ArrayList<>();
+            for (Object object : list) {
+                copyRow.add(object);
+            }
+            copyList.add(copyRow);
+        }
+        return copyList;
+    }
+
+    @Test
+    public void createNewExamTest() throws GeneralSecurityException, InvalidDatabaseDataException, InvalidRequestException, IOException, ExamFormatException {
+
+        List<List<Object>> valueRangeDataWhole = globalValueRangeDataWholeCopy();
+        valueRangeWhole.setValues(valueRangeDataWhole);
+        ExamMetadata actualExamMetadata = new ExamMetadata(spreadsheetId, sheetName, user.getId(), 100);
+        Questions actualQuestions = new Questions();
+        SubjectiveQuestion actualSubjectiveQuestion = new SubjectiveQuestion((String) valueRangeDataWhole.get(2).get(1), 2, 1);
+        List<String> actualOptions = new ArrayList<>();
+        actualOptions.add((String) valueRangeDataWhole.get(3).get(2));
```

Comment:
This is very confusing, and not clear what it is trying to do.

Reply:
Changed this back to base exam format for clearer view listing string only, telling what the field actually represents.